src/opf/socwr.jl

```diff
@@ -2,450 +2,300 @@ struct SOCOPFQuad <: AbstractFormulation end
-            ==
-            0
+        model[:ohm_qt][e] = @constraint(model,
+            branch_status[e] * (-btt[e] * wt[e] - btf[e] * wr[e] - gtf[e] * wi[e]) - qt[e] == 0
         )
-
-        # Voltage angle difference limit
-        model[:voltage_difference_limit_ub][i] = JuMP.@constraint(model, wi_br <= tan(branch["angmax"])*wr_br)
-        model[:voltage_difference_limit_lb][i] = JuMP.@constraint(model, wi_br >= tan(branch["angmin"])*wr_br)
-
-        # Apparent power limit, from side and to side
-        if OPF == SOCOPFQuad
-            model[:thermal_limit_fr][i] = JuMP.@constraint(model, p_fr^2 + q_fr^2 <= branch["rate_a"]^2)
-            model[:thermal_limit_to][i] = JuMP.@constraint(model, p_to^2 + q_to^2 <= branch["rate_a"]^2)
-        elseif OPF == SOCOPF
-            model[:thermal_limit_fr][i] = JuMP.@constraint(model, [branch["rate_a"], p_fr, q_fr] in JuMP.SecondOrderCone())
-            model[:thermal_limit_to][i] = JuMP.@constraint(model, [branch["rate_a"], p_to, q_to] in JuMP.SecondOrderCone())
+    
+        # Thermal limit
+        if OPF == SOCOPF
+            model[:sm_fr][e] = @constraint(model, [smax[e], pf[e], qf[e]] in SecondOrderCone())
+            model[:sm_to][e] = @constraint(model, [smax[e], pt[e], qt[e]] in SecondOrderCone())
+        elseif OPF == SOCOPFQuad
+            model[:sm_fr][e] = @constraint(model, pf[e]^2 + qf[e]^2 ≤ smax[e]^2)
+            model[:sm_to][e] = @constraint(model, pt[e]^2 + qt[e]^2 ≤ smax[e]^2)
         end
-
-        # Jabr inequality (one per branch)
-        if OPF == SOCOPFQuad
-            model[:jabr][i] = JuMP.@constraint(model,
-                wr[i]^2 + wi[i]^2 <= w[branch["f_bus"]]*w[branch["t_bus"]]
-            )
-        elseif OPF == SOCOPF
-            model[:jabr][i] = JuMP.@constraint(model,
-                [w[branch["f_bus"]] / sqrt(2), w[branch["t_bus"]] / sqrt(2), wr[i], wi[i]] in JuMP.RotatedSecondOrderCone()
-            )
+    
+        # Voltage angle difference limit
+        model[:va_diff_lb][e] = @constraint(model, wi[e] >= tan(dvamin[e]) * wr[e])
+        model[:va_diff_ub][e] = @constraint(model, wi[e] <= tan(dvamax[e]) * wr[e])
+
+        # Jabr constraints
+        if OPF == SOCOPF
+            model[:jabr][e] = @constraint(model, [wf[e] / sqrt(2), wt[e] / sqrt(2), wr[e], wi[e]] in RotatedSecondOrderCone())
+        elseif OPF == SOCOPFQuad
+            model[:jabr][e] = @constraint(model, wr[e]^2 + wi[e]^2 ≤ wf[e] * wt[e])
         end
     end
 
     #
     #   III. Objective
     #
-    l, u = extrema(gen["cost"][1] for (i, gen) in ref[:gen])
-    (l == u == 0.0) || @warn "Data $(data["name"]) has quadratic cost terms; those terms are being ignored"
-    JuMP.@objective(model, Min, sum(
-        gen["cost"][2]*pg[i] + gen["cost"][3]
-        for (i,gen) in ref[:gen]
-    ))
-
-    return OPFModel{OPF}(data, model)
-end
-
 
+    @objective(model,
+        Min,
+        sum(c2[g] * pg[g]^2 + c1[g] * pg[g] + c0[g] for g in 1:G if gen_status[g])
```

Comment:
I can see one potential (but hopefully minor) issue with this: JuMP will cast the objective function as quadratic, even if all the quadratic terms are zero. 
If a solver doesn't support quadratic objective, then it will either throw an error or trigger internal bridges, which I'd rather we avoid if possible.

![image](https://github.com/user-attachments/assets/587593a1-8dbb-4296-a0ba-8337280dd291)

There are possible work-arounds, from checking whether we do have non-zero quadratic terms to manually converting to SOC constraints...
If that's OK, I'd prefer we keep support for quadratic objective to a later PR (once we merge #118). That would also give us time to update the math formulations (especially on the duality side)